src/controllers/red-experience.js

```diff
@@ -28,21 +28,21 @@ const redExperienceController = (request) => {
   }
 
   if (request.body.red === 'yes') {
-    // Clean up the user's input before we use it.
+    // Clean up the user's input.
     const {cleanRedExperienceYears, cleanRedControlNumber} = cleanInput(request.body);
     request.session.red = 'yes';
     request.session.redExperienceYears = cleanRedExperienceYears;
     request.session.redControlNumber = cleanRedControlNumber;
 
-    if (request.session.redExperienceYears) {
-      request.session.redExperienceYearsInvalidError = new RegExp(/\D/).test(cleanRedExperienceYears);
+    if (request.body.redExperienceYears) {
+      request.session.redExperienceYearsInvalidError = new RegExp(/\D/).test(request.body.redExperienceYears.trim());
     }
 
-    if (request.session.redControlNumber) {
-      request.session.redControlNumberInvalidError = new RegExp(/\D/).test(cleanRedControlNumber);
+    if (request.body.redControlNumber) {
+      request.session.redControlNumberInvalidError = new RegExp(/\D/).test(request.body.redControlNumber.trim());
     }
 
-    if (request.session.redExperienceYears === '' && request.session.redControlNumber === '') {
+    if (request.body.redExperienceYears.trim() === '' && request.body.redControlNumber.trim() === '') {
       request.session.redEmptyError = true;
     }
```

Comment:
I think this test should be expanded.

Currently it checks if BOTH fields are empty, but then only tells the user about one error.

![image](https://user-images.githubusercontent.com/37802088/95961988-b9cc6800-0dfd-11eb-8594-3a3ce06266a3.png)

If the user then corrects that one error, they're taken on the forward path, not back here to correct the second missing field.

There's also a possibility that `request.body.redExperienceYears` or `request.body.redControlNumber` could be undefined at this point, so you'll get `Cannot read property 'trim' of undefined` errors.

I suggest that this test is split in two and each part is embedded inside the two tests above it.

Reply:
"If the user then corrects that one error, they're taken on the forward path, not back here to correct the second missing field."

So I asked jimmy about the validation on these experience pages as there was a potential scenario where a person may be returning to stalking deer but hasn't controlled any in the last 5 years so wouldn't have to fill out the second box the response I received from him about this is below:

"I would say they can proceed but this should be flagged, no automatic approval"

Reply:
> hasn't controlled any in the last 5 years

There's a difference between 0 and undefined. 0 means 'none', undefined means '🤷🏻‍♀️'. The correct answer for the user here is...

> **What is the average number of red deer you have controlled per year over the last 5 years?**
> `0`


'🤷🏻‍♀️' means you cannot proceed. [We don't ask questions we don't want answers for](https://design-system.service.gov.uk/patterns/question-pages/#when-to-use-this-pattern).

>You should make sure you know why you’re asking every question and only ask users for information you really need.

I's strongly argue against it, but if you really want to make this field optional, GDS has this to say.

> If you ask for optional information, mark the labels of optional fields with ‘(optional)’.